pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [mismatched type in violatesFkConstraintsHelper query]

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/e0d64c03-bbe8-4af3-b0cb-5866cef06f13?testReportRun=8&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [617bf347978dcc0d711399b1a76402d7f88de958](https://github.com/cockroachdb/cockroach/commits/617bf347978dcc0d711399b1a76402d7f88de958):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['6383303776005880229306.14565726'::DECIMAL = '1999-08-20 22:33:07.000795':::TIMESTAMP]) AND count(*) = 0 from schema_w1_11.table_w0_31\n\t    WHERE cȍl31_w0_36 = ('1999-08-20 22:33:07.000795':::TIMESTAMP)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <decimal> = <timestamp> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 113
Total Schema Statements Succeeded = 113
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 67
Total Transactions Rolled Back = 56
Total Transactions Executed = 123
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_8_of_30/test.outputs/logTestWorkload675587457
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload675587457
--- FAIL: TestWorkload (22.44s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36128

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14024830?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14024830?buildTab=artifacts#/) on master @ [7d0697b632066ee78735fc57e8150222d5576d0d](https://github.com/cockroachdb/cockroach/commits/7d0697b632066ee78735fc57e8150222d5576d0d):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [(-2.96182736880654676E+27):::DECIMAL = e'BW~jp\\'\\x0e\\x11R'::NAME:::NAME]) AND count(*) = 0 from public.table_w1_1\n\t    WHERE \"co\\\\xbel1_w1_9\" = (e'BW~jp\\'\\x0e\\x11R'::NAME:::NAME)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY[(-2.96182736880654676E+27):::DECIMAL = e'BW~jp\'\x0e\x11R'::NAME:::NAME]: unsupported comparison operator: <decimal> = <name> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 12
Total Schema Statements Succeeded = 12
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 6
Total Transactions Rolled Back = 9
Total Transactions Executed = 15
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload783464542
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload783464542
--- FAIL: TestWorkload (23.76s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14040432?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14040432?buildTab=artifacts#/) on master @ [1340a1e0419ce5df9ae05f5a362a8bbedbcc0a07](https://github.com/cockroachdb/cockroach/commits/1340a1e0419ce5df9ae05f5a362a8bbedbcc0a07):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['-0.7179127791224806'::FLOAT8 = 'yjvau'::VARCHAR]) AND count(*) = 0 from schema_w1_42.table_w0_32\n\t    WHERE col32_w0_44 = ('yjvau'::VARCHAR)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <float> = <varchar> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 88
Total Schema Statements Succeeded = 88
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 41
Total Transactions Rolled Back = 57
Total Transactions Executed = 98
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload282689912
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload282689912
--- FAIL: TestWorkload (71.93s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14109634?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14109634?buildTab=artifacts#/) on master @ [04f0416d526a43741d22fd03966758dcccdeb79f](https://github.com/cockroachdb/cockroach/commits/04f0416d526a43741d22fd03966758dcccdeb79f):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [ARRAY[(-86.45467643078375079):::DECIMAL,5.663854075949911563E+22:::DECIMAL,(-4.473327843027759606E+24):::DECIMAL,(-58650.36833044207356):::DECIMAL] = ARRAY[(-86.45467643078375079):::DECIMAL,5.663854075949911563E+22:::DECIMAL,(-4.473327843027759606E+24):::DECIMAL,(-58650.36833044207356):::DECIMAL],ARRAY[(-7.595952600872596537E+37):::DECIMAL] = ARRAY[(-86.45467643078375079):::DECIMAL,5.663854075949911563E+22:::DECIMAL,(-4.473327843027759606E+24):::DECIMAL,(-58650.36833044207356):::DECIMAL],ARRAY[(-62828.30572770897655):::DECIMAL,(-6.588003786479220899E+28):::DECIMAL,1.0:::DECIMAL] = ARRAY[(-86.45467643078375079):::DECIMAL,5.663854075949911563E+22:::DECIMAL,(-4.473327843027759606E+24):::DECIMAL,(-58650.36833044207356):::DECIMAL]]) AND count(*) = 0 from schema_w1_41.table_w0_42\n\t    WHERE \"col4*2,_w0_47\" = (ARRAY[(-86.45467643078375079):::DECIMAL,5.663854075949911563E+22:::DECIMAL,(-4.473327843027759606E+24):::DECIMAL,(-58650.36833044207356):::DECIMAL])\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <string> = <decimal[]> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 101
Total Schema Statements Succeeded = 101
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 63
Total Transactions Rolled Back = 72
Total Transactions Executed = 135
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3856467008
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3856467008
--- FAIL: TestWorkload (75.74s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14142888?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14142888?buildTab=artifacts#/) on master @ [24414b864f65c6544ea987f7970fa1c192ca3310](https://github.com/cockroachdb/cockroach/commits/24414b864f65c6544ea987f7970fa1c192ca3310):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['''hITBFxwm''':::TSQUERY = e'\\bP\\x01Y\\x10\\x13)\\x07j'::VARCHAR,'''aZA'' <-> ''K'' & ''VqjCt''':::TSQUERY = e'\\bP\\x01Y\\x10\\x13)\\x07j'::VARCHAR]) AND count(*) = 0 from schema_w1_27.table_w0_60\n\t    WHERE \"\\\\u76FEcol60_w0_65\" = (e'\\bP\\x01Y\\x10\\x13)\\x07j'::VARCHAR)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY[e'\'hITBFxwm\'':::TSQUERY = e'\bP\x01Y\x10\x13)\x07j'::VARCHAR, e'\'aZA\' <-> \'K\' & \'VqjCt\'':::TSQUERY = e'\bP\x01Y\x10\x13)\x07j'::VARCHAR]: unsupported comparison operator: <tsquery> = <varchar> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 93
Total Schema Statements Succeeded = 93
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 48
Total Transactions Rolled Back = 53
Total Transactions Executed = 101
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2441337405
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2441337405
--- FAIL: TestWorkload (68.59s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14148960?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14148960?buildTab=artifacts#/) on master @ [c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb](https://github.com/cockroachdb/cockroach/commits/c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb):


```
    "result": true
   },
   {
    "query": "SELECT ARRAY[parent.table_schema, parent.table_name, parent.column_name, child.column_name] FROM (SELECT conname, conkey, confkey, conrelid, confrelid FROM pg_constraint WHERE (contype = 'f') AND (conrelid = 'schema_w0_21.table_w0_49'::REGCLASS::INT8)) AS con JOIN (SELECT constraint_name FROM information_schema.table_constraints WHERE ((table_schema = 'schema_w0_21') AND (table_name = 'table_w0_49')) AND (crdb_internal.is_constraint_active('schema_w0_21.table_w0_49', constraint_name) = true)) AS tc ON conname = tc.constraint_name JOIN (SELECT column_name, ordinal_position, column_default FROM information_schema.columns WHERE (table_schema = 'schema_w0_21') AND (table_name = 'table_w0_49')) AS child ON conkey[1] = child.ordinal_position JOIN (SELECT pc.oid, cols.table_schema, cols.table_name, cols.column_name, cols.ordinal_position FROM pg_class AS pc JOIN pg_namespace AS pn ON pc.relnamespace = pn.oid JOIN information_schema.columns AS cols ON ((pc.relname = cols.table_name) AND (pn.nspname = cols.table_schema))) AS parent ON ((con.confkey[1] = parent.ordinal_position) AND (con.confrelid = parent.oid)) WHERE child.column_name != 'rowid'",
    "result": null
   }
  ],
  "previousStatements": [
   "CREATE SCHEMA IF NOT EXISTS schema_w1_46",
   "INSERT INTO schema_w0_21.table_w0_49 (col49_w0_50,\" co%pl49_w0_51\",\"c%vol49 _w0_52\",\"cOl49_w0_53\",\"co\u000cl49_w0_54\",\"co l49_w0_55\",col49_w0_56,col49_w0_57,\"col\t49_w0_58\",col49_w0_59) VALUES ('d9733985-fb3d-45c7-a2f2-c7e20602273a':::UUID,'1995-05-18':::DATE,(-8):::INT8,'01060000E0E61000000800000001030000C0010000000D0000006A76BF5D6DCC63C080DBB802E81439C01441EC0DAEFCE4C10A638A4E160AFBC1F5E6B73BE94363C091279F1CE81C44C00269295FE527F6413CC8503A6108E4C1B73F39CE6CEB50C05B5241DE263755C0E84C783716B3E741FEF080E3EE4DEAC1EA175CD1085B5240126F8D01F2F854C040E87F0B0917B541A400D3E8047AF5C1DAD3C277665A6040CE7F161B23DB50C0F4F0457ECF94FEC1F64EA487504AFA41E8C311751555664000E8353617C3DABF2CEAAFF8348AE1C1BF74E572EDCDF0C1F80682D689B85140EC89974F24693440E6A142869530E6C17848968A290CF141D4EE4182A8F35E403C1C70C40B4044407CF44F9CD7B2E941E4D1A62D2C78E541D03709E7681D20C06C60FE7763F53B402E41E4E693C7F0419422ED5C6BFCF241F8880D3F9FAC51C008442868278D4440D4D7B464B830F541D0D548EC8369DAC1386FF23E346261C08C44B94D69375640C2301700BFE7014218CEE2E23747E1419F8D15E74E2C62C0F45F0E7E9D4D33403BB868B54474FAC1FD094927634AFEC16A76BF5D6DCC63C080DBB802E81439C01441EC0DAEFCE4C10A638A4E160AFBC101030000C0010000000A000000416BCF72200862C080C63B2A1CD51D4070896F757EEBCDC15ACD6D2EE4E4F54162A3DF20D48A60C0804413911432104058821E9E7222FE4120FC1EFF3D62D24159E3337911DD55C05062BE0BF8AB1AC0B89DF0D82AEFF0C1A2311E88ED60F9C105151F052CE45AC0DAFA36CB59F655C0B8279C78884A0142B88A0EF4195FFDC180D41F22F56E37407CD2D2BE99DA31403CCFF50128F5FA41208201BAB153E941EA5FC56FD68D61406A40643535A541402EB4A3C9BD51E5C1ECC7382853BDF8C156CB68C99487634030A2B8E1B1A8534028F4E59A0705E8C166D54F56058D01428079CF29C03CF03FCAC33B4FBC325240807272C6AA1493C1BAD74FAB014FECC182CF2012D8B35FC076CA115ED04753409094242E1952C4C1707D4B4146320042416BCF72200862C080C63B2A1CD51D4070896F757EEBCDC15ACD6D2EE4E4F54101030000C0010000000A0000003523A6F348B061C0C08B07A4F5CF28409068223B4D8BDF417264A5B49D7501C2675CE5F49B0866C0B89B87C63A0D27C09090F679CE61DFC1B010DDCCABA4ED4100FE2BD0455B48C0C829645E2E8236C054CD696150F1DAC122A86AA3190CF141486C92613D6F3AC08E4141ADFE874FC0709946B78256F541CC44F6413B2FFE41AC29F42C3EC96040D82F14A7B5F14540BA132CDEFFB2FBC1082A55587DB6E1C1862A607D7E0E4BC038B83A3EE901354000A3A26ABE39B3417CD37CADE0C6E041ECCF43E1315A43C0C086F5293F034D40A02AE9061725B841F05EFDC48B4BE9C1A3A0B9CD265961C0C61FB72EE51D5040F841F099D635D341BCE4374E1EB6FC415F879B5A54FE55C0E82510784DB33A4080FB7293665792C1369653FCE51502423523A6F348B061C0C08B07A4F5CF28409068223B4D8BDF417264A5B49D7501C201030000C0010000000B000000D8201AFD42AF5AC01C6CF58860AC45C0F4D7EAE6CECAEBC17CBF25D11D9AF8C18EBA98A9487F62C04233499C9EF94EC062DCD790E79AF8C162AE67751701F4C1C2619DA765DF4EC06633427ACA074CC04EE3BB4772A400C2F813BF54040FDA417450864A7BFE454020C547FE9F6A53C0A108A2B6ED0701C28E9F804A2B19F741E8A39F51AD4F6240C4973142A40433C0F0D1719DE6B2D7C1A09EBB839A90C341EAF434618C94604038382F6DF58D31404965D311D4E7F9C1F6AF46749CFDE6C1845291242E724C40A097BB440D7917404EB3903FC35D01422F56D018DF7DFFC1584145F36C0C5440E4161D9254BF46403A61948D0829EBC1C2ADCAAB259FF1C11061D56B572533409E5826CEEB2B514081F596EC9A9300C2D4ED5F29679BF64186C92E402C4D60C0A2AA3382367F534000C7FBA54755F741E0FB98D8B412EF41D8201AFD42AF5AC01C6CF58860AC45C0F4D7EAE6CECAEBC17CBF25D11D9AF8C101030000C001000000060000001C6793B9DE8461C00A5598838A3752C0AC1209478D98F9C1C0804A573532ADC10082F527FAB6F43F304FA195132551C04D568CD0B55A02C2D4069804ED00F841B05A9A67FC1B6340553B3719A9C153C0BAB85E2B5A19004280DEBA6878B4C4C144917B1E1CC75B400085243A43AA02C05CE2241E9E050042202A232847E0F7C1147EDB0BF9DC47401066AAB3B75A4B40750663ABFF1EFEC130FD2A840D59DCC11C6793B9DE8461C00A5598838A3752C0AC1209478D98F9C1C0804A573532ADC101030000C001000000050000007375F4534CA054C01468D070732F3EC0F46DB759F4EFF8C1A0620799A5ABF4418C5DEF9B9ECD51C0DA99C44A930041C060FF9049D19FF641A67774BEFB7DE1C1E027B716C92B24C00CF2C7AA528538C0828456E16B61EBC1947D4343B14BF641305E4DF0498341C0E07A54A5572C23C06CBA24A7F655E3415803B0DEA4E0E3C17375F4534CA054C01468D070732F3EC0F46DB759F4EFF8C1A0620799A5ABF44101030000C0010000000B00000087B0A92FF40260C0748EE358E1443FC0E055FA5E7E4BFDC13A075E9AB87FF341008C14206A6261405ABA06C7DAD44EC00EDF8AF4E06AFAC17EE546DFFA6D02422CA6D97DF66E5B40386C24A5C1772CC0F89768803BEDE141503D6F8C9CD1C8C104196A97E01E65408499CE2D45893A4080ED6BE5FC5CF6C1CD070A65862DF2C118B3D9B899E63740B04EFE9A7AD844405F3217BDAF1BFCC152910AFA773CF44120C6D982B9202540B0C9CD5AFBE836404244FE0DB862F94130BE8DD7D0DFC5C17031D738A67E2A4012DA4101BC63514050920D1D5E3EEF410465B5EAB00501C200686F5B35A4B9BF38B85232B4F54D40801F2CBCA917C44112D6B0A0D50FFDC1651A2D8D5CFC5BC00C506C966B874740DA49F33111A6F5C169854149767100C20A5B39DFEB9660C0F8A6FDE560FC4440BEA476E9FF3DE9C160C744518C77C9C187B0A92FF40260C0748EE358E1443FC0E055FA5E7E4BFDC13A075E9AB87FF34101030000C0010000000A000000E90818BF1F7850C0F388F65AA76047C0302B82BCEA6EFC4140500377C104A7C142C37F95A6ED50C0F3064EEFCFFB51C0C05674BBE7A8F3416C497D09F2F0DDC1083EBF8FA4BB3DC020D364BF148614C025C8C8641421F4C12EF4E88CC2BBF3C1FC72AADB254D4A40270DB8405F2249C08549267828C8FDC1A47591312123DCC1F45C32CFB33E4040804406AADA60F73F7875A371BCD6FDC18CF00951192AE941BC73996E31984040F08C053E1C03344020453778298DF341DC5AAF063FEAFD419A3929416F0B504048C0159ABF325240E84A46BA1220F54100CD43635EFDEE410DE9B817836157C0D6787A567B0E504030AA3B5FA695CF416A958C9C8D52F241A56B563F223B62C0844C9BF5A0615140165FD40B459CE6C1650C1055AC32FFC1E90818BF1F7850C0F388F65AA76047C0302B82BCEA6EFC4140500377C104A7C1':::GEOGRAPHY,'1995-03-08 21:31:33.000362+00':::TIMESTAMPTZ,'00:00:00':::TIME,'?QIt\u003c'::STRING,e'hx;l\\x0fcK3':::REFCURSOR,(-1.4237642288208008):::FLOAT8,'20 years 1 mon 179 days 19:54:19.50648':::INTERVAL),('ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID,'1991-06-06':::DATE,61:::INT8,'01060000E0E61000000200000001030000C00100000007000000AFA0A2F51AED5AC04ADD9A8043B843C0086793785679DAC15D7C214A6A73F8C1E996BF3D045551C0579AC654E74252C094E650FE5443EC4100CFE6D22A28D2411C4743363F5562404833F4499B7C27C070D35A8E68ADC1C130BCC9A9CE9A00C2B41F60B77EE66240E0867BC3088819C040D38CB11389B54194B5B55FC432E1C108C37932EE4E584090802907AF321D40BD2409F2C293FDC1A4688DD4BB52ECC158508EC87A3B6640F48FAA40DA1251409A7777B86686024214D235F22303EC41AFA0A2F51AED5AC04ADD9A8043B843C0086793785679DAC15D7C214A6A73F8C101030000C00100000004000000EB02967C437765C09852DDB5C66D3040F0A2453FA59CEFC133CA1EE6F39301C2B86FA7E83C9F38C058FBF1FBFBBD2740D4E7101AA720EF41CFAF241ACA97F4C1AD219E37F56962C05CE02A24B0DF37408217F4B31B2BE7C1888FA0B67F19FF41EB02967C437765C09852DDB5C66D3040F0A2453FA59CEFC133CA1EE6F39301C2':::GEOGRAPHY,'294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ,'02:20:20.906666':::TIME,e'\\x06K\\x12\\x11p'::STRING,'7':::REFCURSOR,0.2880493402481079:::FLOAT8,'-81 years -8 mons -656 days -11:42:51.689754':::INTERVAL)"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: integer out of range (SQLSTATE 22003)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3122
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: integer out of range (SQLSTATE 22003)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 67
Total Schema Statements Succeeded = 67
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 38
Total Transactions Rolled Back = 54
Total Transactions Executed = 92
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload23076323
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload23076323
--- FAIL: TestWorkload (53.68s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.1 release-blocker]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/e71abb08-5df1-4731-8696-95790f0b9cf2?testReportRun=14&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb](https://github.com/cockroachdb/cockroach/commits/c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [1114:::OID = e'~\\x01h\\x1bGN'::STRING]) AND count(*) = 0 from schema_w0_6.table_w0_22\n\t    WHERE \"C o\"\"l\\\\U000318FD22_w0_34\" = (e'~\\x01h\\x1bGN'::STRING)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY[1114:::OID = e'~\x01h\x1bGN'::STRING]: unsupported comparison operator: <oid> = <string> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 58
Total Schema Statements Succeeded = 58
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 35
Total Transactions Rolled Back = 34
Total Transactions Executed = 69
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_14_of_30/test.outputs/logTestWorkload4128754569
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload4128754569
--- FAIL: TestWorkload (12.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=14</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.1 release-blocker]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14155396?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14155396?buildTab=artifacts#/) on master @ [140ecc0c19c46204bb04a70ea0e513c45a26ad0d](https://github.com/cockroachdb/cockroach/commits/140ecc0c19c46204bb04a70ea0e513c45a26ad0d):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['12:57:12.869434-13:56':::TIMETZ = '12:57:12.869434-13:56':::TIMETZ]) AND count(*) = 0 from schema_w0_1.table_w0_2\n\t    WHERE \"co%l2_w0_6\" = ('12:57:12.869434-13:56':::TIMETZ)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <string> = <timetz> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 16
Total Schema Statements Succeeded = 16
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 6
Total Transactions Rolled Back = 6
Total Transactions Executed = 12
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2982225966
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2982225966
--- FAIL: TestWorkload (28.36s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.1 release-blocker]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14168214?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14168214?buildTab=artifacts#/) on master @ [7bb52a7d1c75d5adfdfa53e5fcff6f5e6497408f](https://github.com/cockroachdb/cockroach/commits/7bb52a7d1c75d5adfdfa53e5fcff6f5e6497408f):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['010100008087BB2A8BEA14FFC166F3E3D50A33004246E28CFD435502C2':::GEOMETRY = '010100008087BB2A8BEA14FFC166F3E3D50A33004246E28CFD435502C2':::GEOMETRY,'01020000C0050000009B0D80B0B99A02C2F9FC3AC7A3D2FAC1E6423AFD7405FA41C85230F11D340142503D73BD9B5501C259FA06A8B9E000C268FF51A6D35901427C110A67D382FBC168E3278F101EFB41A0BCA9688CEFBC4160473C14A8E1D941D78C583957D6FFC1D0129A06CBD6E341E044B60B3E82B0414A251CF950AFF641401A2514498ED0C1AA382D22D5BDF74140410B3E8A74F341F01536EF0F9EC8C1789941E1B438FD41':::GEOMETRY = '010100008087BB2A8BEA14FFC166F3E3D50A33004246E28CFD435502C2':::GEOMETRY,'0103000000010000000500000000000000000000000000000000000000000000000000F03F0000000000000000000000000000F03F000000000000F03F0000000000000000000000000000F03F00000000000000000000000000000000':::GEOMETRY = '010100008087BB2A8BEA14FFC166F3E3D50A33004246E28CFD435502C2':::GEOMETRY]) AND count(*) = 0 from schema_w0_24.table_w0_40\n\t    WHERE \"col4(0_w0_57\" = ('010100008087BB2A8BEA14FFC166F3E3D50A33004246E28CFD435502C2':::GEOMETRY)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <string> = <geometry> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 98
Total Schema Statements Succeeded = 98
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 59
Total Transactions Rolled Back = 52
Total Transactions Executed = 111
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3917684841
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3917684841
--- FAIL: TestWorkload (82.23s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.1 release-blocker]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14184570?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14184570?buildTab=artifacts#/) on master @ [fed794450894a7d1197c763ed26f75d773b93070](https://github.com/cockroachdb/cockroach/commits/fed794450894a7d1197c763ed26f75d773b93070):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [3220:::OID = 'cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET,26:::OID = 'cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET,16:::OID = 'cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET]) AND count(*) = 0 from schema_w0_54.table_w0_62\n\t    WHERE \"col 62\t_w0_68\" = ('cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY[3220:::OID = 'cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET, 26:::OID = 'cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET, 16:::OID = 'cdcb:5962:69f4:7e06:c010:da0a:25fa:9fa4/5':::INET]: unsupported comparison operator: <oid> = <inet> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 152
Total Schema Statements Succeeded = 152
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 85
Total Transactions Rolled Back = 93
Total Transactions Executed = 178
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2090896781
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2090896781
--- FAIL: TestWorkload (111.83s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14194529?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14194529?buildTab=artifacts#/) on master @ [14d19acfdfbc8eb5074ddb38f29ed92c40fac35b](https://github.com/cockroachdb/cockroach/commits/14d19acfdfbc8eb5074ddb38f29ed92c40fac35b):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['1974-04-08 08:57:31.000076':::TIMESTAMP = '1974-04-08 08:57:31.000076':::TIMESTAMP,'1970-06-16 05:44:46.000509':::TIMESTAMP = '1974-04-08 08:57:31.000076':::TIMESTAMP,'1977-04-22 21:32:52.000684':::TIMESTAMP = '1974-04-08 08:57:31.000076':::TIMESTAMP]) AND count(*) = 0 from schema_w0_1.table_w0_10\n\t    WHERE \"c'ol10_w0_19\" = ('1974-04-08 08:57:31.000076':::TIMESTAMP)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <time> = <timestamp> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 30
Total Schema Statements Succeeded = 30
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 12
Total Transactions Rolled Back = 25
Total Transactions Executed = 37
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3757018903
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3757018903
--- FAIL: TestWorkload (34.74s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14205135?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14205135?buildTab=artifacts#/) on master @ [d4a51117deac22a2ffa280bac5404811cd694265](https://github.com/cockroachdb/cockroach/commits/d4a51117deac22a2ffa280bac5404811cd694265):


```
     ]
    ]
   },
   {
    "query": "SELECT (false = ANY (ARRAY[4905080500.501836859:::DECIMAL = 4905080500.501836859:::DECIMAL])) AND (count(*) = 0) FROM public.table_w1_6 WHERE \"c\"\"ol6\\\\xb9_w1_10\" = (4905080500.501836859:::DECIMAL)",
    "result": false
   }
  ],
  "previousStatements": [
   "INSERT INTO public.table_w1_6 (col6_w1_7,col6_w1_8,col6_w1_9) VALUES (4905080500.501836859:::DECIMAL,'193.161.226.13/3':::INET,'}'::\"char\")"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: insert on table "table_w1_6" violates foreign key constraint "table_w1_6_c""ol6\\xb9_w1_10_table_w1_6_col6_w1_7_fk" (SQLSTATE 23503)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3122
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: insert on table "table_w1_6" violates foreign key constraint "table_w1_6_c""ol6\\xb9_w1_10_table_w1_6_col6_w1_7_fk" (SQLSTATE 23503)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 43
Total Schema Statements Succeeded = 43
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 18
Total Transactions Rolled Back = 25
Total Transactions Executed = 43
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload307612601
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload307612601
--- FAIL: TestWorkload (40.76s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14215387?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14215387?buildTab=artifacts#/) on master @ [791969fa98711e681289fa4f61f3d58c8fe44263](https://github.com/cockroachdb/cockroach/commits/791969fa98711e681289fa4f61f3d58c8fe44263):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [false = false,true = false,false = false]) AND count(*) = 0 from schema_w1_2.table_w0_21\n\t    WHERE \"c\"\"😛ol21_w0_38\" = (false)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <string> = <bool> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 65
Total Schema Statements Succeeded = 65
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 32
Total Transactions Rolled Back = 45
Total Transactions Executed = 77
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3701511083
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3701511083
--- FAIL: TestWorkload (48.26s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/b300b368-e0fb-4f3b-ab26-9e3f16e52fee?testReportRun=10&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [7f45cb61df77419ded37619f701993ee410f6951](https://github.com/cockroachdb/cockroach/commits/7f45cb61df77419ded37619f701993ee410f6951):


```
   {
    "query": "SELECT ARRAY[parent.table_schema, parent.table_name, parent.column_name, child.column_name] FROM (SELECT conname, conkey, confkey, conrelid, confrelid FROM pg_constraint WHERE (contype = 'f') AND (conrelid = 'schema_w0_29.table_w0_69'::REGCLASS::INT8)) AS con JOIN (SELECT constraint_name FROM information_schema.table_constraints WHERE ((table_schema = 'schema_w0_29') AND (table_name = 'table_w0_69')) AND (crdb_internal.is_constraint_active('schema_w0_29.table_w0_69', constraint_name) = true)) AS tc ON conname = tc.constraint_name JOIN (SELECT column_name, ordinal_position, column_default FROM information_schema.columns WHERE (table_schema = 'schema_w0_29') AND (table_name = 'table_w0_69')) AS child ON conkey[1] = child.ordinal_position JOIN (SELECT pc.oid, cols.table_schema, cols.table_name, cols.column_name, cols.ordinal_position FROM pg_class AS pc JOIN pg_namespace AS pn ON pc.relnamespace = pn.oid JOIN information_schema.columns AS cols ON ((pc.relname = cols.table_name) AND (pn.nspname = cols.table_schema))) AS parent ON ((con.confkey[1] = parent.ordinal_position) AND (con.confrelid = parent.oid)) WHERE child.column_name != 'rowid'",
    "result": null
   }
  ],
  "previousStatements": [
   "ALTER TABLE schema_w0_29.table_w0_69 ADD CONSTRAINT \"table_w0_69_col69_w0_79_table_w0_69_c\fo̟l😡69_w0_73_fk\" FOREIGN KEY (\"c\fo̟l😡69_w0_73\") REFERENCES schema_w0_29.table_w0_69 (col69_w0_79) ON DELETE CASCADE ON UPDATE CASCADE",
   "ALTER FUNCTION schema_w0_7.udf_w0_30(IN pg_catalog.int8, IN pg_catalog.int8) SET SCHEMA schema_w1_37",
   "SELECT t0.\"col6\\n9_w0_75\" AS col0, t0.col69_w0_74 AS col1, t0.\"c-ol69_w0_76\" AS col2 FROM schema_w0_29.table_w0_69 AS t0 FETCH FIRST 1 ROWS ONLY",
   "INSERT INTO schema_w0_29.table_w0_69 (\"col 😡69_w0_70\",\"co'l69\\\\U000CBA59_w0_71\",col69_w0_72,\"c\fo̟l😡69_w0_73\",col69_w0_74,\"col6\\n9_w0_75\",\"c-ol69_w0_76\",\"'col69_w0_77\",col69_w0_78,col69_w0_79,col69_w0_80,col69_w0_81) VALUES (0.0:::FLOAT8,'2015-05-20 21:45:12.000994':::TIMESTAMP,1043:::OID,'2019-04-23 01:29:10.00083+00':::TIMESTAMPTZ,'0104000060E61000000800000001010000400E9C813E169A63C01643DCB02C9C41C086662F3CD812EDC10101000040EC59939ADAA54840523C7392F88D514094A65812FEF3F1410101000040D4BFDED3BAE449C041FCE3D5EB4953C093E007F5617AF8C10101000040D4C43EFAEE814640B432B69B065935408059FD29EFC4C1410101000040B0B1919B40B146C0C87CADC6C8AC50400020595135FA7C410101000040801464E40C4EF2BFD65C74D5D2D14CC0884F4F5C16E101420101000040996A580339DE5FC068DC1AB344742FC086698657778BF3410101000040C8D43B8629F56140D0E5C485D79D4A4064417B56B02FED41':::GEOGRAPHY,6.209732203235692571E+32:::DECIMAL,B'00000000000000000000000000000000000000000',e'\\x1ca\\x16p/\\x10L':::REFCURSOR,true,'1998-05-11 11:11:40.000755+00':::TIMESTAMPTZ,ARRAY['\\x96':::BYTES,'\\x85190b40':::BYTES,'\\x':::BYTES,'\\xe415':::BYTES,'\\x2a44':::BYTES],'e'::\"char\"),((-0.27818015217781067):::FLOAT8,'2018-03-16 21:20:55.000293':::TIMESTAMP,2205:::OID,'2005-06-30 17:13:20.000926+00':::TIMESTAMPTZ,'0107000020E6100000080000000104000020E61000000700000001010000005A81D2F487F764C0D01ECA04473333C00101000000982B9FF782DD36C04CDFB263C120524001010000000EDBF198BDE261C08A7C01F6114A55C00101000000F846F3B551845BC08ED5C844493F4CC00101000000D0616A7DC78246C0AF52D782D7F04BC00101000000A0D28ACC5D535240F849D787075848400101000000B81D3E9855AD4640A892DC1815533A400105000020E61000000700000001020000000200000018D28D3FDA876040178AA532A0CF54C03CFAD14372D441C09C16F01927D05140010200000002000000DA3D8E9E84C661C088A9680629914CC0DEE5168E1D8A5AC0E8B66598C8DA52C00102000000050000005801F9BD427F4D406E0A5B95435655C0A014257BCE905C405BE6FB5FF64755C0E41798902AE05A40C09DC3C5289F51C0CCF1190D4646664060153B9F7FC729404899A9EB03B66340CA28BEC5B3EE5340010200000005000000BF76BD988F9552C01C79662BE72F4CC0D89BB33E9ABB5F4028AF6227956F51C0086F04EEE457514098136081AF5C30403CDD74BE5E7C454024EB29F3D1924440D8F0AF91B91E48C0A85310B1F8A229C00102000000040000005C98D57751064D406045D4CF888830C0DCBDD12D099D5D40C032681DD9731EC0BE5F216232E465C02A160B2E798C534068257A33097560C078F0B2648471484001020000000200000010725325CF0D5A40886628C36E3221C0BC3137DD343A63404A6FB5CD0DD65140010200000005000000D7D9E2EC538D65C0C2ADD9BF7FF74DC0D05B94AD43BB3740A8CBAA8B8D0443C00CE3C45108794440C884B5EAEB5C45C0E85C64B5ACA14C403106DFDDEB9541C0247AF6A2ECEE6340A9A7A03146A652C00103000020E610000001000000070000006840B8CA1CF165C0445FC0AFD3E848C050D659BAB68235C0130F5A6F7E1355C02CE8ED2733814A402C1182A0A8E93440186ABE1E45D842C0E21064917DC4414084E69E34F2235EC0D06E28C6B6004A403F8E8BAF673F5DC078B6C31983C02B406840B8CA1CF165C0445FC0AFD3E848C00103000020E61000000100000005000000E43B8FE70BFC474004C859B4578235C08C7DEFE03AAD5E40581BBBE41D7144408CECB31698AA5B40F42095382FF94A406E0A3CF5B9A154C04EF831B42FE55540E43B8FE70BFC474004C859B4578235C00104000020E61000000300000001010000005802ECCF71AC4EC058F8D19ADCD221C00101000000A0CF5DF929EA364030B8994FED9F3DC001010000004E75FEE819E354C0862FAD7A780B50400104000020E6100000060000000101000000A0380BDE91F85340D8D6536ED74232C00101000000D448760DE0815C40D6785806165946C00101000000C81017ADC4CC37C05345649E33D94CC00101000000246E3042944140C02A99BBB1CEE655C00101000000A03EAB10D7A02D4080D4E335C0892A40010100000096B82D6E8B9C51C09A71B1A0069F4FC00104000020E6100000090000000101000000BA727D6DC66263C03A29527E6C1155C00101000000401B41B30CCF3B4000A8841C3EADE53F0101000000EE427C2CA0C963C0C8392C47BC5A514001010000008643C89CFACA6440364F2A67F4B653C0010100000020AD5E524AF365408463954EE66331400101000000E8CE507257B844C01453F870C691364001010000006CAD6B48216756C0D357E4EFB1C74EC0010100000050FE5718BA0262C0E9D7A3DFC5B74BC001010000008814B46FE2364940DC968ABC47AD38C00103000020E61000000100000006000000BCA4C22D05C95340DE8CABED85064FC05CAAE667BE995F4056D59124168E54C01C80270B27BE6240CE6EEA3FECBF4EC0A2C09BF468F360409CAC3C7765234440DEA3BF14FF8A5DC0E89B2EF93FD94440BCA4C22D05C95340DE8CABED85064FC0':::GEOGRAPHY,4740140339.701063905:::DECIMAL,B'11110000111111001100100010101001011100110',e'\\x19e4':::REFCURSOR,false,'2010-09-11 15:44:56.000373+00':::TIMESTAMPTZ,ARRAY['\\x62':::BYTES,'\\xb63a784ae6c632':::BYTES,'\\xd69c699dde84a3':::BYTES],e'\\U00002603'::\"char\"),(1.0:::FLOAT8,'2020-09-24 01:16:16.000446':::TIMESTAMP,1043:::OID,'1978-11-01 07:00:51.000229+00':::TIMESTAMPTZ,'0106000020E61000000200000001030000000100000004000000000000000000444000000000000044400000000000003440000000000080464000000000008046400000000000003E4000000000000044400000000000004440010300000002000000060000000000000000003440000000000080414000000000000024400000000000003E40000000000000244000000000000024400000000000003E4000000000000014400000000000804640000000000000344000000000000034400000000000804140040000000000000000003E40000000000000344000000000000034400000000000002E40000000000000344000000000000039400000000000003E400000000000003440':::GEOGRAPHY,(-2405220840044.965691):::DECIMAL,B'00111011110011100110101111000010010011111',e'qu\\x16\\x13mI':::REFCURSOR,true,'1976-03-11 22:34:26.000473+00':::TIMESTAMPTZ,ARRAY['\\x':::BYTES,'\\xc7872098076749f67a':::BYTES,'\\x':::BYTES],'i'::\"char\")"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: insert on table "table_w0_69" violates foreign key constraint "table_w0_69_col69_w0_79_table_w0_69_c�o̟l😡69_w0_73_fk" (SQLSTATE 23503)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3122
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: insert on table "table_w0_69" violates foreign key constraint "table_w0_69_col69_w0_79_table_w0_69_c�o̟l😡69_w0_73_fk" (SQLSTATE 23503)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 144
Total Schema Statements Succeeded = 144
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 84
Total Transactions Rolled Back = 70
Total Transactions Executed = 154
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_10_of_30/test.outputs/logTestWorkload1785391357
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1785391357
--- FAIL: TestWorkload (24.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14221544?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14221544?buildTab=artifacts#/) on master @ [7f45cb61df77419ded37619f701993ee410f6951](https://github.com/cockroachdb/cockroach/commits/7f45cb61df77419ded37619f701993ee410f6951):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['''OmkHmezk'' ''eIq'' ''m'' ''pBbXBCKu''':::TSVECTOR = e'\\tsrx\\x18?rQ'::STRING,'''M'':418B,570A,692A ''Q'' ''XGX'':718C ''YUzETann'':90A,424B,992A ''ZWWmlHdzlt'':442C,552B ''wVsPI'' ''znP'':189B,450B,718C':::TSVECTOR = e'\\tsrx\\x18?rQ'::STRING,'''BClfaio'':311C,381A,389B,955B':::TSVECTOR = e'\\tsrx\\x18?rQ'::STRING]) AND count(*) = 0 from schema_w1_4.table_w1_19\n\t    WHERE \"c%pol19_w1_30\" = (e'\\tsrx\\x18?rQ'::STRING)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY[e'\'OmkHmezk\' \'eIq\' \'m\' \'pBbXBCKu\'':::TSVECTOR = e'\tsrx\x18?rQ'::STRING, e'\'M\':418B,570A,692A \'Q\' \'XGX\':718C \'YUzETann\':90A,424B,992A \'ZWWmlHdzlt\':442C,552B \'wVsPI\' \'znP\':189B,450B,718C':::TSVECTOR = e'\tsrx\x18?rQ'::STRING, e'\'BClfaio\':311C,381A,389B,955B':::TSVECTOR = e'\tsrx\x18?rQ'::STRING]: unsupported comparison operator: <tsvector> = <string> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 74
Total Schema Statements Succeeded = 74
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 35
Total Transactions Rolled Back = 40
Total Transactions Executed = 75
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload693964335
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload693964335
--- FAIL: TestWorkload (71.63s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14227820?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14227820?buildTab=artifacts#/) on master @ [51bbfff84c26be8a2b40e25b4bce3d59ea63dc59](https://github.com/cockroachdb/cockroach/commits/51bbfff84c26be8a2b40e25b4bce3d59ea63dc59):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [(-96):::INT8 = '4 years 1 mon 46 days 05:52:56.821882':::INTERVAL]) AND count(*) = 0 from public.table_w0_19\n\t    WHERE col19_w0_29 = ('4 years 1 mon 46 days 05:52:56.821882':::INTERVAL)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <int2> = <interval> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 54
Total Schema Statements Succeeded = 54
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 27
Total Transactions Rolled Back = 40
Total Transactions Executed = 67
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload1962749230
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload1962749230
--- FAIL: TestWorkload (10.36s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14228136?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14228136?buildTab=artifacts#/) on master @ [51bbfff84c26be8a2b40e25b4bce3d59ea63dc59](https://github.com/cockroachdb/cockroach/commits/51bbfff84c26be8a2b40e25b4bce3d59ea63dc59):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY [(-15):::INT8 = '!''ZtRUBllnU'' & ''yYYkTuLA'' <-> ''OcoZf''':::TSQUERY,28:::INT8 = '!''ZtRUBllnU'' & ''yYYkTuLA'' <-> ''OcoZf''':::TSQUERY]) AND count(*) = 0 from schema_w0_36.table_w1_64\n\t    WHERE col64_w1_72 = ('!''ZtRUBllnU'' & ''yYYkTuLA'' <-> ''OcoZf''':::TSQUERY)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <int2> = <tsquery> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 132
Total Schema Statements Succeeded = 132
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 65
Total Transactions Rolled Back = 88
Total Transactions Executed = 153
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2715339797
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2715339797
--- FAIL: TestWorkload (95.62s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14258859?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14258859?buildTab=artifacts#/) on master @ [ce5f34ea97475f45fa354e58aacf424779d0de49](https://github.com/cockroachdb/cockroach/commits/ce5f34ea97475f45fa354e58aacf424779d0de49):


```
     "schema_w1_31",
     "schema_w0_43",
     "schema_w0_45"
    ]
   }
  ],
  "previousStatements": [
   "ALTER TYPE public.enum_w1_16 DROP VALUE 'ssdsf'",
   "CREATE SEQUENCE IF NOT EXISTS schema_w0_38.seq_w0_48",
   "DROP SCHEMA \"schema_w0_6\" CASCADE",
   "CREATE FUNCTION udf_w0_49(IN p1 public.enum_w1_16)\n\tRETURNS VOID\n\tLANGUAGE SQL\n\tAS $$ SELECT NULL $$"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3137
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 72
Total Schema Statements Succeeded = 72
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 41
Total Transactions Rolled Back = 68
Total Transactions Executed = 109
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2049577533
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2049577533
--- FAIL: TestWorkload (69.13s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/dd5d9389-79fc-4f60-8f28-5be60f21ca0c?testReportRun=11&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [f4dc2b59541e11cf06e2b948c7c37bf1aed21868](https://github.com/cockroachdb/cockroach/commits/f4dc2b59541e11cf06e2b948c7c37bf1aed21868):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['BOX(-0.7656181783207502 0.10829422925978194,0.04496271613178948 0.38948458378188155)':::BOX2D = 90002:::OID,'BOX(-1.3240758689296883 -0.5529358154331204,0.25315012421447514 0.6386364168933412)':::BOX2D = 90002:::OID]) AND count(*) = 0 from schema_w0_6.table_w1_15\n\t    WHERE \"col1 5_w1_18\" = (90002:::OID)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY['BOX(-0.7656181783207502 0.10829422925978194,0.04496271613178948 0.38948458378188155)':::BOX2D = 90002:::OID, 'BOX(-1.3240758689296883 -0.5529358154331204,0.25315012421447514 0.6386364168933412)':::BOX2D = 90002:::OID]: unsupported comparison operator: <box2d> = <oid> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 45
Total Schema Statements Succeeded = 45
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 28
Total Transactions Rolled Back = 42
Total Transactions Executed = 70
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_11_of_30/test.outputs/logTestWorkload3774306415
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload3774306415
--- FAIL: TestWorkload (12.54s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=11</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14276390?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14276390?buildTab=artifacts#/) on master @ [f4dc2b59541e11cf06e2b948c7c37bf1aed21868](https://github.com/cockroachdb/cockroach/commits/f4dc2b59541e11cf06e2b948c7c37bf1aed21868):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2921
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:196
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['01040000c00700000001010000c0f490261587d901420008194659f200c267867fcc729a01c224a6a5417206014201010000c0206b89860f6be54188d5d559ac8100c230aedd67159ffec12a63083cb2fbe0c101010000c084ded8c8516a01c22823db25d7e0fa4137ae85f9972df5c138a233ef9c33fd4101010000c0904707e22a18d0414ac2ef06c093e2c170a4174f0090c7c1a66dfabc200100c201010000c02ed053d522cc00c28037382cc0b59a41f6ca7bde955af5415cd961c6744ef14101010000c070366913d0abf7c1c632dd41afdcf5c1600b3faae1ecb041642d33e7386df1c101010000c000d44955c9a1b541e055bc865810edc1f80fdcefc3a1f8413eb0882b413bfa41':::STRING = e'\\U000849FC\\U000B82BD\\U000D8212\\U000D3214\\U000E82C0\\U000E8EA3\\U00017DBD\\U0007AD39' COLLATE da_DK,'01040000c00700000001010000c0f490261587d901420008194659f200c267867fcc729a01c224a6a5417206014201010000c0206b89860f6be54188d5d559ac8100c230aedd67159ffec12a63083cb2fbe0c101010000c084ded8c8516a01c22823db25d7e0fa4137ae85f9972df5c138a233ef9c33fd4101010000c0904707e22a18d0414ac2ef06c093e2c170a4174f0090c7c1a66dfabc200100c201010000c02ed053d522cc00c28037382cc0b59a41f6ca7bde955af5415cd961c6744ef14101010000c070366913d0abf7c1c632dd41afdcf5c1600b3faae1ecb041642d33e7386df1c101010000c000d44955c9a1b541e055bc865810edc1f80fdcefc3a1f8413eb0882b413bfa41':::STRING = e'\\U000849FC\\U000B82BD\\U000D8212\\U000D3214\\U000E82C0\\U000E8EA3\\U00017DBD\\U0007AD39' COLLATE da_DK]) AND count(*) = 0 from schema_w1_3.table_w1_4\n\t    WHERE col4_w1_17 = (e'\\U000849FC\\U000B82BD\\U000D8212\\U000D3214\\U000E82C0\\U000E8EA3\\U00017DBD\\U0007AD39' COLLATE da_DK)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <string> = <collatedstring{da_dk}> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 13
Total Schema Statements Succeeded = 13
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 9
Total Transactions Rolled Back = 16
Total Transactions Executed = 25
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload248743567
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload248743567
--- FAIL: TestWorkload (29.12s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14307427?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14307427?buildTab=artifacts#/) on master @ [72646a555214c0705781e440b9df585d5eea9511](https://github.com/cockroachdb/cockroach/commits/72646a555214c0705781e440b9df585d5eea9511):


```
     "schema_w1_52",
     "schema_w1_55",
     "schema_w0_29",
     "schema_w1_60"
    ]
   }
  ],
  "previousStatements": [
   "ALTER TYPE public.enum_w0_18 DROP VALUE 'saff'",
   "SELECT 'validating all objects', crdb_internal.validate_multi_region_zone_configs()",
   "CREATE FUNCTION udf_w0_35(IN p1 public.enum_w0_18)\n\tRETURNS VOID\n\tLANGUAGE SQL\n\tAS $$ SELECT NULL $$"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3137
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 101
Total Schema Statements Succeeded = 101
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 50
Total Transactions Rolled Back = 95
Total Transactions Executed = 145
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3223274831
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload3223274831
--- FAIL: TestWorkload (64.28s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14328561?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14328561?buildTab=artifacts#/) on master @ [2e7ab27484135f6cd16199004ba9f20dce777859](https://github.com/cockroachdb/cockroach/commits/2e7ab27484135f6cd16199004ba9f20dce777859):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2922
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:197
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['BOX(-10 -10,10 10)':::BOX2D = B'1','BOX(-0.48495319978470036 -1.3660511892107063,0.4012508230611492 0.7218770515286164)':::BOX2D = B'1']) AND count(*) = 0 from schema_w0_40.table_w1_80\n\t    WHERE col80_w1_87 = (B'1')\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <box2d> = <varbit> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 142
Total Schema Statements Succeeded = 142
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 82
Total Transactions Rolled Back = 114
Total Transactions Executed = 196
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2972359829
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2972359829
--- FAIL: TestWorkload (132.50s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/9aaaff38-1c4f-437d-a1a8-7cd889ced196?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [431d56840cacdc86638ccb48e9f3bdc16897d341](https://github.com/cockroachdb/cockroach/commits/431d56840cacdc86638ccb48e9f3bdc16897d341):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:454
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2922
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:197
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:441
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['4714-11-24 BC':::DATE = '4714-11-24 BC':::DATE,'1982-12-24':::DATE = '4714-11-24 BC':::DATE,'1986-03-09':::DATE = '4714-11-24 BC':::DATE]) AND count(*) = 0 from schema_w1_16.table_w0_45\n\t    WHERE \"co\fl45_w0_59\" = ('4714-11-24 BC':::DATE)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: <string> = <date> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 67
Total Schema Statements Succeeded = 67
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 37
Total Transactions Rolled Back = 49
Total Transactions Executed = 86
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_3_of_30/test.outputs/logTestWorkload2157339838
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload2157339838
--- FAIL: TestWorkload (15.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14348348?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14348348?buildTab=artifacts#/) on master @ [431d56840cacdc86638ccb48e9f3bdc16897d341](https://github.com/cockroachdb/cockroach/commits/431d56840cacdc86638ccb48e9f3bdc16897d341):


```
     "schema_w1_46",
     "schema_w0_58",
     "schema_w0_62",
     "schema_w1_64"
    ]
   }
  ],
  "previousStatements": [
   "DROP SEQUENCE schema_w0_51.seq_w0_64",
   "ALTER TYPE public.enum_w1_58 DROP VALUE 's'",
   "CREATE FUNCTION udf_w0_66(IN p1 public.enum_w1_58)\n\tRETURNS VOID\n\tLANGUAGE SQL\n\tAS $$ SELECT NULL $$"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3138
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 132
Total Schema Statements Succeeded = 132
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 71
Total Transactions Rolled Back = 112
Total Transactions Executed = 183
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload4294388477
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload4294388477
--- FAIL: TestWorkload (84.02s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119621 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/acd782ea-c80f-4c0b-a0c5-d65d3183b37f?testReportRun=27&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [455b16592df7d8efd121b3ba1256fb477e227564](https://github.com/cockroachdb/cockroach/commits/455b16592df7d8efd121b3ba1256fb477e227564):


```
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:495
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.CollectOne[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:167
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.Scan[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/query_util.go:149
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).scanBool
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:88
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraintsHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1202
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).violatesFkConstraints
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/error_screening.go:1121
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).insertRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:2922
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:197
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:482
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:586
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (6) CollectOne: Query: "\n\t    SELECT false = ANY (ARRAY ['19:37:14.808015+06:11':::TIMETZ = e'7?,\\b['::STRING,'17:33:11.4988+02:26':::TIMETZ = e'7?,\\b['::STRING,'01:32:39.57083+14:52':::TIMETZ = e'7?,\\b['::STRING]) AND count(*) = 0 from schema_w0_4.table_w0_14\n\t    WHERE \"c ol1 4_w0_31\" = (e'7?,\\b['::STRING)\n\t" []
        	            	Wraps: (7) ERROR: unsupported comparison operator: false = ANY ARRAY['19:37:14.808015+06:11':::TIMETZ = e'7?,\b['::STRING, '17:33:11.4988+02:26':::TIMETZ = e'7?,\b['::STRING, '01:32:39.57083+14:52':::TIMETZ = e'7?,\b['::STRING]: unsupported comparison operator: <timetz> = <string> (SQLSTATE 22023)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 46
Total Schema Statements Succeeded = 46
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 27
Total Transactions Rolled Back = 31
Total Transactions Executed = 58
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_27_of_30/test.outputs/logTestWorkload315216621
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload315216621
--- FAIL: TestWorkload (13.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=27</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/workload/schemachange/error_screening.go
+++ b/pkg/workload/schemachange/error_screening.go
@@ -257,7 +257,7 @@ func (og *operationGenerator) valuesViolateUniqueConstraints(
 	ctx context.Context,
 	tx pgx.Tx,
 	tableName *tree.TableName,
-	columns []string,
+	nonGeneratedColNames []string,
 	colInfo []column,
 	rows [][]string,
 ) (bool, codesWithConditions, error) {
@@ -276,8 +276,8 @@ func (og *operationGenerator) valuesViolateUniqueConstraints(
 		hasGenerationError := false
 		// Put values to be inserted into a column name to value map to simplify lookups.
 		columnsToValues := map[string]string{}
-		for i := 0; i < len(columns); i++ {
-			columnsToValues[columns[i]] = row[i]
+		for i := 0; i < len(nonGeneratedColNames); i++ {
+			columnsToValues[nonGeneratedColNames[i]] = row[i]
 		}
 		newCols := make(map[string]string)
 		// Resolve any generated expressions, which have been validated earlier.
@@ -533,7 +533,7 @@ func (og *operationGenerator) validateGeneratedExpressionsForInsert(
 	ctx context.Context,
 	tx pgx.Tx,
 	tableName *tree.TableName,
-	columns []string,
+	nonGeneratedColNames []string,
 	colInfos []column,
 	row []string,
 ) (
@@ -544,8 +544,8 @@ func (og *operationGenerator) validateGeneratedExpressionsForInsert(
 ) {
 	// Put values to be inserted into a column name to value map to simplify lookups.
 	columnsToValues := map[string]string{}
-	for i := 0; i < len(columns); i++ {
-		columnsToValues[columns[i]] = row[i]
+	for i := 0; i < len(nonGeneratedColNames); i++ {
+		columnsToValues[nonGeneratedColNames[i]] = row[i]
 	}
 	nullViolationAdded := false
 	validateExpression := func(expr string, typ string, isNullable bool, addGenerated bool) error {
@@ -1059,21 +1059,25 @@ func (og *operationGenerator) checkAndAdjustForUnknownSchemaErrors(err error) er
 
 // violatesFkConstraints checks if the rows to be inserted will result in a foreign key violation.
 func (og *operationGenerator) violatesFkConstraints(
-	ctx context.Context, tx pgx.Tx, tableName *tree.TableName, columns []string, rows [][]string,
+	ctx context.Context,
+	tx pgx.Tx,
+	tableName *tree.TableName,
+	nonGeneratedColNames []string,
+	rows [][]string,
 ) (bool, error) {
+	// TODO(annie): readd the join on active constraints once #120702 is resolved.
+	//
+	// N.B. We add random noise to column names that makes it hard to just directly call on these names. This is
+	// not the case with table/schema names; thus, only column names are quote_ident'ed to ensure that they get
+	// referenced properly.
 	fkConstraints, err := og.scanStringArrayRows(ctx, tx, fmt.Sprintf(`
-		SELECT array[parent.table_schema, parent.table_name, parent.column_name, child.column_name]
+		SELECT array[parent.table_schema, parent.table_name, quote_ident(parent.column_name), quote_ident(child.column_name)]
 		  FROM (
 		        SELECT conname, conkey, confkey, conrelid, confrelid
 		          FROM pg_constraint
 		         WHERE contype = 'f'
 		           AND conrelid = '%s'::REGCLASS::INT8
 		       ) AS con
-			JOIN ( SELECT CONSTRAINT_NAME from information_schema.table_constraints 
-				      WHERE table_schema ='%s' AND
-								    table_name='%s' AND
-										(crdb_internal.is_constraint_active('%s', constraint_name) = true)
-           ) AS tc ON conname = tc.CONSTRAINT_NAME
 		  JOIN (
 		        SELECT column_name, ordinal_position, column_default
 		          FROM information_schema.columns
@@ -1094,7 +1098,7 @@ func (og *operationGenerator) violatesFkConstraints(
 		                       AND con.confrelid = parent.oid
 		                      )
 		 WHERE child.column_name != 'rowid';
-`, tableName.String(), tableName.Schema(), tableName.Object(), tableName.String(), tableName.Schema(), tableName.Object()))
+`, tableName.String(), tableName.Schema(), tableName.Object()))
 	if err != nil {
 		return false, og.checkAndAdjustForUnknownSchemaErrors(err)
 	}
@@ -1102,9 +1106,11 @@ func (og *operationGenerator) violatesFkConstraints(
 	// Maps a column name to its index. This way, the value of a column in a row can be looked up
 	// using row[colToIndexMap["columnName"]] = "valueForColumn"
 	columnNameToIndexMap := map[string]int{}
-	for i, name := range columns {
+
+	for i, name := range nonGeneratedColNames {
 		columnNameToIndexMap[name] = i
 	}
+
 	for _, row := range rows {
 		for _, constraint := range fkConstraints {
 			parentTableSchema := constraint[0]
@@ -1142,12 +1148,16 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 	parentTableSchema, parentTableName, parentColumn, childColumn string,
 	childTableName *tree.TableName,
 	parentAndChildAreSameTable bool,
-	row []string,
+	rowToInsert []string,
 	allRows [][]string,
 ) (bool, error) {
 
+	childIndex, ok := columnNameToIndexMap[childColumn]
+	if !ok {
+		return false, errors.Newf("child column %s does not exist in table %s", childColumn, childTableName)
+	}
+	childValue := rowToInsert[childIndex]
 	// If the value to insert in the child column is NULL and the column default is NULL, then it is not possible to have a fk violation.
-	childValue := row[columnNameToIndexMap[childColumn]]
 	if childValue == "NULL" {
 		return false, nil
 	}
@@ -1159,28 +1169,38 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 		if err != nil {
 			return false, err
 		}
-		// Put values to be inserted into a column name to value map to simplify lookups.
-		columnsToValues := map[string]string{}
-		for name, idx := range columnNameToIndexMap {
-			columnsToValues[name] = row[idx]
-		}
-		colIdx := 0
-		for idx, colInfo := range colsInfo {
+
+		var parentColInfo *column
+		for _, colInfo := range colsInfo {
 			if colInfo.name == parentColumn {
-				colIdx = idx
+				parentColInfo = &colInfo
 				break
 			}
 		}
+		if parentColInfo == nil {
+			return false, errors.Newf("column %s not found in columns for %s", parentColumn, childTableName)
+		}
+
 		for _, otherRow := range allRows {
-			parentValueInSameInsert := otherRow[columnNameToIndexMap[parentColumn]]
-			// If the parent column is generated, spend time to generate the value.
-			if colsInfo[colIdx].generated {
-				var err error
-				parentValueInSameInsert, err = og.generateColumn(ctx, tx, colsInfo[colIdx], columnsToValues)
+			var parentValueInSameInsert string
+			if parentColInfo.generated {
+				// If the parent column is a computed column, spend time to generate the value.
+				columnsToValues := map[string]string{}
+				for name, idx := range columnNameToIndexMap {
+					columnsToValues[name] = rowToInsert[idx]
+				}
+				parentValueInSameInsert, err = og.generateColumn(ctx, tx, colsInfo[parentColInfo.ordinal], columnsToValues)
 				if err != nil {
 					return false, err
 				}
+			} else {
+				parentIdx, ok := columnNameToIndexMap[parentColumn]
+				if !ok {
+					return false, errors.Newf("parent column %s does not exist in table %s", parentColumn, childTableName)
+				}
+				parentValueInSameInsert = otherRow[parentIdx]
 			}
+
 			// Skip over NULL values.
 			if parentValueInSameInsert == "NULL" {
 				continue
@@ -1198,7 +1218,7 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 	    SELECT %s count(*) = 0 from %s.%s
 	    WHERE %s = (%s)
 	`,
-		checkSharedParentChildRows, parentTableSchema, parentTableName, tree.NameString(parentColumn), childValue)
+		checkSharedParentChildRows, parentTableSchema, parentTableName, parentColumn, childValue)
 	return og.scanBool(ctx, tx, q)
 }
 

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -887,9 +887,6 @@ func (og *operationGenerator) addForeignKeyConstraint(
 	stmt.potentialExecErrors.add(pgcode.ForeignKeyViolation)
 	og.potentialCommitErrors.add(pgcode.ForeignKeyViolation)
 
-	// TODO why did I add this??
-	stmt.potentialExecErrors.add(pgcode.FeatureNotSupported)
-
 	// It's possible for the table to be dropped concurrently, while we are running
 	// validation. In which case a potential commit error is an undefined table
 	// error.
@@ -2845,10 +2842,10 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 		}
 		nonGeneratedCols = truncated
 	}
-	colNames := []string{}
+	nonGeneratedColNames := []string{}
 	rows := [][]string{}
 	for _, col := range nonGeneratedCols {
-		colNames = append(colNames, col.name)
+		nonGeneratedColNames = append(nonGeneratedColNames, col.name)
 	}
 	numRows := og.randIntn(3) + 1
 	for i := 0; i < numRows; i++ {
@@ -2884,7 +2881,7 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 	anyInvalidInserts := false
 	stmt = makeOpStmt(OpStmtDML)
 	for _, row := range rows {
-		invalidInsert, generatedErrors, potentialErrors, err := og.validateGeneratedExpressionsForInsert(ctx, tx, tableName, colNames, allColumns, row)
+		invalidInsert, generatedErrors, potentialErrors, err := og.validateGeneratedExpressionsForInsert(ctx, tx, tableName, nonGeneratedColNames, allColumns, row)
 		if err != nil {
 			return nil, err
 		}
@@ -2910,7 +2907,7 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 		// Verify if the new row will violate unique constraints by checking the constraints and
 		// existing rows in the database.
 		var generatedErrors codesWithConditions
-		uniqueConstraintViolation, generatedErrors, err = og.valuesViolateUniqueConstraints(ctx, tx, tableName, colNames, allColumns, rows)
+		uniqueConstraintViolation, generatedErrors, err = og.valuesViolateUniqueConstraints(ctx, tx, tableName, nonGeneratedColNames, allColumns, rows)
 		if err != nil {
 			return nil, err
 		}
@@ -2919,7 +2916,7 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 		}
 		// Verify if the new row will violate fk constraints by checking the constraints and rows
 		// in the database.
-		fkViolation, err = og.violatesFkConstraints(ctx, tx, tableName, colNames, rows)
+		fkViolation, err = og.violatesFkConstraints(ctx, tx, tableName, nonGeneratedColNames, rows)
 		if err != nil {
 			return nil, err
 		}
@@ -2943,7 +2940,7 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 	stmt.sql = fmt.Sprintf(
 		`INSERT INTO %s (%s) VALUES %s`,
 		tableName,
-		strings.Join(colNames, ","),
+		strings.Join(nonGeneratedColNames, ","),
 		strings.Join(formattedRows, ","),
 	)
 	return stmt, nil
@@ -3190,6 +3187,7 @@ type column struct {
 	nullable            bool
 	generated           bool
 	generatedExpression string
+	ordinal             int
 }
 
 func (og *operationGenerator) getTableColumns(
@@ -3209,14 +3207,16 @@ func (og *operationGenerator) getTableColumns(
                       ),
          columns AS (
                   SELECT c->>'computeExpr' AS generation_expression,
-                         c->>'name' AS column_name
+                         c->>'name' AS column_name,
+												 c->>'id' AS ordinal
                     FROM columns_json
                  )
   SELECT quote_ident(show_columns.column_name),
          show_columns.data_type,
          show_columns.is_nullable,
          columns.generation_expression IS NOT NULL AS is_generated,
-         COALESCE(columns.generation_expression, '') as generated_expression
+         COALESCE(columns.generation_expression, '') AS generated_expression,
+			   columns.ordinal::INT-1
     FROM [SHOW COLUMNS FROM %s] AS show_columns, columns
    WHERE show_columns.column_name != 'rowid'
          AND show_columns.column_name = columns.column_name
@@ -3232,7 +3232,7 @@ func (og *operationGenerator) getTableColumns(
 	for rows.Next() {
 		var c column
 		var typName string
-		err := rows.Scan(&c.name, &typName, &c.nullable, &c.generated, &c.generatedExpression)
+		err := rows.Scan(&c.name, &typName, &c.nullable, &c.generated, &c.generatedExpression, &c.ordinal)
 		if err != nil {
 			return nil, err
 		}

--- a/pkg/workload/schemachange/optype.go
+++ b/pkg/workload/schemachange/optype.go
@@ -265,7 +265,7 @@ var opWeights = []int{
 	alterFunctionRename:               1,
 	alterFunctionSetSchema:            1,
 	alterTableAddColumn:               1,
-	alterTableAddConstraintForeignKey: 0, // Disabled due to #119288.
+	alterTableAddConstraintForeignKey: 1,
 	alterTableAddConstraintUnique:     0,
 	alterTableAlterColumnType:         0, // Disabled and tracked with #66662.
 	alterTableAlterPrimaryKey:         1,

